src/Features/Core/Portable/Completion/CompletionServiceWithProviders.cs

```diff
@@ -48,7 +48,19 @@ protected virtual ImmutableArray<CompletionProvider> GetBuiltInProviders()
 
         protected ImmutableArray<CompletionProvider> GetProviders(ImmutableHashSet<string> roles)
         {
-            return _providerSource.GetProviders(roles ?? ImmutableHashSet<string>.Empty);
+            var getProviderTask = _providerSource.GetProvidersAsync(roles ?? ImmutableHashSet<string>.Empty);
+            if (getProviderTask.IsCompleted)
+            {
+                return getProviderTask.Result;
+            }
+
+            return ImmutableArray<CompletionProvider>.Empty;
```

Comment:
1. this absolutely needs to be documented.
2. i really don't like this.  this just means users will try to get completion and will get nothing.

Reply:
I have added a timeout to mitigate this.